py4j-python/src/py4j/java_gateway.py

```diff
@@ -1598,8 +1601,9 @@ def start(self):
             socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
         try:
             self.server_socket.bind((self.address, self.port))
+            self.__listening_address, self.__listening_port = self.server_socket.getsockname()
         except Exception as e:
-            msg = "An error occurred while trying to start the callback server"
+            msg = "An error occurred while trying to start the callback server (%s:%d)" % (self.address, self.port)
```

Comment:
this is a very good idea. I should print these info as often as possible. Please use .format though.